libtransmission/utils.h

```diff
@@ -63,6 +63,14 @@ std::optional<std::locale> tr_locale_set_global(std::locale const& locale) noexc
 
 bool tr_file_read(std::string_view filename, std::vector<char>& contents, tr_error* error = nullptr);
 
+/**
+ * Tries to move a file by renaming, but never by copying.
+ */
+bool tr_file_move_strict(std::string_view oldpath_in, std::string_view newpath_in, tr_error* error = nullptr);
```

Comment:
Is this new function needed? Can't callers just use `tr_sys_path_rename()` instead?

Reply:
The main difference is that this new function creates intermediary directories if they are missing. I'll make the doxygen more explicit.